sunpy/map/sources/trace.py

```diff
@@ -55,6 +55,20 @@ def __init__(self, data, header, **kwargs):
         self.plot_settings['norm'] = ImageNormalize(
             stretch=source_stretch(self.meta, LogStretch()), clip=False)
 
+    @property
+    def coordinate_system(self):
+        """
+        Override the default implementation to handle TRACEMAP-specific logic for CTYPE values.
+        """
+        if self.meta:
+            ctype1 = self.meta.get('ctype1', '')
+            ctype2 = self.meta.get('ctype2', '')
+            if ctype1.lower() in ("solar-x", "solar_x"):
+                ctype1 = 'HPLN-TAN'
+            if ctype2.lower() in ("solar-y", "solar_y"):
+                ctype2 = 'HPLT-TAN'
+            return SpatialPair(ctype1, ctype2)
```

Comment:
```suggestion
        ctype1, ctype2 = self.meta['ctype1'], self.meta['ctype2']
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```
Same here: this overridden method ought to error if `meta` does not have `ctype1` or `ctype2`.